echo_server.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+import socket
+import os
+import mimetypes
+
+def response_ok(t):
+    okay = u"HTTP/1.1 200 OK\r\n"
+    uritype = u"Content-Type: " + t[0] + "\r\n"
+    c_len = u"Content-length: " + str(len(t[1]))
+    crlf = u"\r\n\r\n"
+    body = t[1]
+    return "{}{}{}{}{}".format(okay, uritype, c_len, crlf, body).encode('utf-8')
+
+def response_error(error_code, reason):
+    return u"HTTP/1.1 {} {}\r\n \
```

Comment:
using the line continuation character here is going to make a real mess of your error response.  Note that you have an extra space after the `"\r\n"` that ends the first line of the response.  Where does that space show up?  How about the extra spaces at the start of the line with the `Content-Type` header?  

Protocols are picky and specific.  You should be just as picky and specific when implementing them.  Remember that computers are very very literal and will not make intelligent, independent decisions about how to parse your work.